### ISSUE TITLE
Chatwoot v4 beta release: Discussion 

### ISSUE THREAD

**Issue by pranavrajs** [chatwoot/chatwoot-mobile-app#862]
A thread to discuss the issue with Chatwoot v4 app release.

If you like to test the latest app, please sign up to the beta testing. 

**iOS**: https://testflight.apple.com/join/yQ4yoSx4
**Android**: https://play.google.com/apps/testing/com.chatwoot.app

Issues: 

- [x] #863

**Comment by pranavrajs** [comment]
@dlc-letelier A new build added to the testflight build, it should be available after the review.

**Comment by dlc-letelier** [comment]
@pranavrajs , thanks for your answer. 

Just installed. I note than, when the connection goes down, the messages did not disappear (very good). But, if I update the screen while the app is offline, swiping down, all conversations disappears (maybe the option to swipe down has to be disable when there is no connection). 

And also, when the app is offline, it keeps appearing two messages saying almost the same: one in the upper part ('You must connect...'), and the other, in the bottom ('Could not connect...'). The one in the bottom dissappears after a while, but keeps appearing every time you enter a conversation.



![image](https://github.com/user-attachments/assets/d47c6d8c-0c42-40e0-911f-04b9e9f7d8fb)

**Comment by pranavrajs** [comment]
> But, if I update the screen while the app is offline, swiping down, all conversations disappears (maybe the option to swipe down has to be disable when there is no connection).

Yes, this is a good feedback. It doesn't need to remove the existing conversations. 

> And also, when the app is offline, it keeps appearing two messages saying almost the same: one in the upper part ('You must connect...'), and the other, in the bottom ('Could not connect...'). The one in the bottom dissappears after a while, but keeps appearing every time you enter a conversation.

Makes sense, this can be updated. These are 2 different toasts. We could remove one of them.

**Comment by raymao9** [comment]
In the testlight app, the Inbox name doesn’t show on conversation interface, and the agent’s name in chinese has been cut at the setting page.

**Comment by muhsin-k** [comment]
@raymao9 

> Inbox name doesn’t show on conversation interface

Thanks for the feedback. We will discuss with team and will priortise it.

> agent’s name in chinese has been cut at the setting page.

Could you please send the screenshot?

**Comment by raymao9** [comment]
> @raymao9
> 
> > Inbox name doesn’t show on conversation interface
> 
> Thanks for the feedback. We will discuss with team and will priortise it.
> 
> > agent’s name in chinese has been cut at the setting page.
> 
> Could you please send the screenshot?

![IMG_0821](https://github.com/user-attachments/assets/5084868c-5eea-4979-b70d-7682b333dc92)

**Comment by dlc-letelier** [comment]
@pranavrajs, now the toast is fine. The problem is that if I close the app, I lose all the conversations. I understood there is a new store for this info. It will possible to access it offline?

**Comment by JaideepGuntupalli** [comment]
@muhsin-k General query about the planned v4 mobile app release: Do we have to upgrade our self-hosted chatwoot backend to latest version when v4 of mobile app is made live? Also what are the timelines for making new version of the app live?

**Comment by muhsin-k** [comment]
> @muhsin-k General query about the planned v4 mobile app release: Do we have to upgrade our self-hosted chatwoot backend to latest version when v4 of mobile app is made live? Also what are the timelines for making new version of the app live?

No, the minimum supported version is 3.13.0.

**Comment by muhsin-k** [comment]
>  It will possible to access it offline?

@dlc-letelier We don't have any plan to support offline mode at the moment.

**Comment by hereck** [comment]
Tem previsão de permitir envio de audio via app?

**Comment by williamjduarte** [comment]
Congratulations on the new app, it's more fluid.

I tested the audio playback feature and it worked. One suggestion would be to allow audio playback at 1.5x and 2x, like on WhatsApp. It's an incredible feature for customer service.

Is there any plan for the audio recording feature on the app?

**Comment by muhsin-k** [comment]
>
> I tested the audio playback feature and it worked. One suggestion would be to allow audio playback at 1.5x and 2x, like on WhatsApp. It's an incredible feature for customer service.
> 

We have a plan to get the similar behaviour.

> Is there any plan for the audio recording feature on the app?

Yes. We are planning to add this in the upcoming releases. Tracking over [here](https://github.com/chatwoot/chatwoot-mobile-app/issues/757).

**Comment by muhsin-k** [comment]
> Issues Persisting in Chatwoot Mobile App (Version 4.0.0)
> 
> Hello, I've tested the latest Chatwoot Mobile App version 4.0.0, but the issue with attaching photos is still not fixed. Additionally, a new issue has appeared regarding text input in the chat. Below are the details of both problems:
> 
> 1. Photo Attachment Issue (Still Present) – Previously Reported in Issue [Cannot Attach Photos in Chatwoot Mobile App (Android 1.10.50) – Missing File Access Permission #870](https://github.com/chatwoot/chatwoot-mobile-app/issues/870)
> 
> * This issue was previously described in issue [Cannot Attach Photos in Chatwoot Mobile App (Android 1.10.50) – Missing File Access Permission #870](https://github.com/chatwoot/chatwoot-mobile-app/issues/870).
> * When trying to attach a photo, the app requests file access permissions.
> * However, there is no option to grant storage or file access permissions in the app settings.
> * Clearing cache, reinstalling, and checking system settings does not resolve the issue.
> 
> 2. Text Input Scrolling Issue (New Bug in 4.0.0)
> 
> * When typing a long message in the chat, the input field does not allow vertical scrolling.
> * If the text exceeds the visible area, there is no way to scroll up to review or edit the earlier part of the message.
> * Additionally, tapping to move the cursor to another part of the text does not work. The only way to navigate is by using the spacebar on the keyboard.
> 
> Environment Details Deployment: A self hosted installation of Chatwoot Device Name: Xiaomi Poco F3 OS Version: Android 13 Chatwoot App Version: 4.0.0 (The latest public version from Google Play) Chatwoot Server Version: 3.14.1
> 
> Both of these issues significantly impact usability.

This issue has been fixed in version 4.0.6. Please update your app through the Google Play Store and try again.

**Comment by raymao9** [comment]
> [@raymao9](https://github.com/raymao9)
> 
> > Inbox name doesn’t show on conversation interface
> 
> Thanks for the feedback. We will discuss with team and will priortise it.
> 
> > agent’s name in chinese has been cut at the setting page.
> 
> Could you please send the screenshot?

Hi, this issue has not been fixed yet, but I saw the v4.0.0 version with new interface has been put on the app store?

**Comment by rapps01** [comment]
I uninstalled version 4.0.6 and re-downloaded 1.10.51 on iOS. The old version has no Chinese display issue and the inbox name can be seen on the conversation list. I have multiple inboxes so this function is much important to me, but you remove it at new version (why?), the new interface is simple and good looking but with no information. I hope this useful function can be retained in the new version.

**Comment by muhsin-k** [comment]
> The old version has no Chinese display issue and the inbox name can be seen on the conversation list

This is fixed. Please check the v4.0.9.

> I have multiple inboxes so this function is much important to me, but you remove it at new version (why?), the new interface is simple and good looking but with no information. I hope this useful function can be retained in the new version.

Regarding the inbox name, we removed it since it was taking up too much space in the card. We completely agree that it's difficult to distinguish between multiple inboxes of the same type. What is your opinion on using an inbox icon instead of the general icon?

**Comment by muhsin-k** [comment]
> Quick Reply Scroll Issue
In the chat, when using "/" to open the quick reply window, scrolling does not work.
If the list of quick replies is long, only the first 4 responses are visible (on my device), and there is no way to scroll to see the rest.
"Delete" Button Not Working in Chat
When trying to delete a message in the chat, the "Delete" button does not respond.
Clicking the button does nothing - there is no error message or any indication of action being performed.

@zsky01 All these issues are fixed. Please check the v4.0.9.

**Comment by muhsin-k** [comment]
Closing this since we will track all the issues via individual issues.

**Comment by rapps01** [comment]
> [@rapps01](https://github.com/rapps01)
> 
> > The old version has no Chinese display issue and the inbox name can be seen on the conversation list
> 
> This is fixed. Please check the v4.0.9.
> 
> > I have multiple inboxes so this function is much important to me, but you remove it at new version (why?), the new interface is simple and good looking but with no information. I hope this useful function can be retained in the new version.
> 
> Regarding the inbox name, we removed it since it was taking up too much space in the card. We completely agree that it's difficult to distinguish between multiple inboxes of the same type. What is your opinion on using an inbox icon instead of the general icon?

I agree that long inbox names might take up too much space, maybe it would be a good idea to add short text or an icon to each inbox to display in the chat interface?
